### PR TITLE
fix secret plugin to not allow empty username

### LINKED ISSUE #1461: Secret plugin doesn't check for the values for the username and server variables to be non-empty
**Bug description**
Secret plugin allows empty username or server values. it doesn't check for the username and server values to be non-empty.

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [X] Addons
- [X] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [X] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behaviour**
Secret plugin should ensure the provided username and server values are non-empty. We should see the following behaviour.:
```
export testuesr=""

tanzu secret registry add test-secret --username=$testuesr --password="test-pass" --server=""
Error: the value for --server flag should not be empty
Error: exit status 1

tanzu secret registry add test-secret --username=$testuesr --password="test-pass" --server="test-server"
Error: the value for --username flag should not be empty
Error: exit status 1

export testuesr=test-user
tanzu secret registry add test-secret --username=$testuesr --password="pass" --server="server"
| Adding registry secret 'test-secret'...
 Added registry secret 'test-secret' into namespace 'default'
```

**Steps to reproduce the bug**
```
export testuesr=""
tanzu secret registry add test-secret --username=$testuesr --password="pass" --server=""
| Adding registry secret 'test-secret'...
 Added registry secret 'test-secret' into namespace 'default'

This behaviour has caused secrets with empty values for username and server flags to be created and causes updating those to fail with the following:

tanzu secret registry update test-secret --username="test-user" --password="pass"
\ Updating registry secret 'test-secret'... 
\ Updating registry secret 'test-secret'... Error: no 'username' entry exists in secret 'test-secret-2'
Error: exit status 1
```

**Version** (include the SHA if the version is not obvious)
```
tanzu version
version: v0.15.0-dev
buildDate: 2022-01-07
sha: 1984c0bf-dirty
```

**Environment where the bug was observed (cloud, OS, etc)**
TKG 1.5 rc.5 cluster

**Relevant Debug Output (Logs, manifests, etc)**
Mentioned in the "Steps to reproduce the bug"

### PR DESCRIPTION
**Pull request by maralavi**
### What this PR does / why we need it
This PR fixes secret plugin to not allow empty username or server.
Closes https://github.com/vmware-tanzu/tanzu-framework/issues/1461

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

Tested on a local cluster

### Release note

```release-note
This PR fixes secret plugin to not allow empty username or server
```

### PR Checklist

- [X] Squash the commits into one or a small number of logical commits
      
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by shivaani0505 — COMMENTED** [review]
Changes look good. You might have to fix cyclomatic complexity lint issue

File: cmd/cli/plugin/secret/registry_secret_add.go [hunk 1]
```diff
@@ -56,14 +57,9 @@ func init() {
 func registrySecretAdd(cmd *cobra.Command, args []string) error {
 	registrySecretOp.SecretName = args[0]
 
-	password, err := extractPassword()
-	if err != nil {
+	if err := validateSecretCredentials(); err != nil {
```
Comment (maralavi) [inline cmd/cli/plugin/secret/registry_secret_add.go:60]: Defined a new funstion due to linter gocyclo error

File: pkg/v1/tkg/tkgpackageclient/registry_secret_update.go [hunk 2]
```diff
@@ -25,54 +25,28 @@ var (
...
 	if err := p.kappClient.GetClient().Get(context.Background(), crtclient.ObjectKey{Name: o.SecretName, Namespace: o.Namespace}, secret); err != nil {
 		if apierrors.IsNotFound(err) {
 			return fmt.Errorf("secret '%s' does not exist in namespace '%s'", o.SecretName, o.Namespace)
 		}
 		return err
 	}
 
-	secretToUpdate := secret.DeepCopy()
-	if err := json.Unmarshal(secretToUpdate.Data[corev1.DockerConfigJsonKey], &dataMap); err != nil {
+	registry, username, password, err := extractExistingSecretCredentials(secret, o)
```
Comment (maralavi) [inline pkg/v1/tkg/tkgpackageclient/registry_secret_update.go:35]: Defined a new funstion due to linter gocyclo error

File: cmd/cli/plugin/secret/registry_secret_add.go [hunk 3]
```diff
@@ -141,6 +137,27 @@ func updateSecret(cmd *cobra.Command, pkgClient tkgpackageclient.TKGPackageClien
 	return nil
 }
 
+func validateSecretCredentials() error {
```
Comment (blc1996) [inline cmd/cli/plugin/secret/registry_secret_add.go:140]: Nit pick: Would it be better to name it `fetchAndValidateSecretCredentials`?

**Review by blc1996 — COMMENTED** [review]
LGTM

File: cmd/cli/plugin/secret/registry_secret_add.go (hunk 3 above)
Comment (maralavi) [reply cmd/cli/plugin/secret/registry_secret_add.go]: Done

**Review by danniel1205 — APPROVED** [review]
LGTM

**Pull request merged**